lattice-controller/src/main.rs

```diff
@@ -51,290 +54,267 @@ impl Default for ConnectionConfig {
-        //   If multiple actors use the same credentials (or multiple actors use no credentials),
-        //   a single nats connection could be shared by multiple actors ids,
-        //   and the index in the HashMap could be hash(creds) instead of actor id.
-        //   Since there aren't many actors linked to this provider, it's simpler
-        //   to follow the pattern used by other capability providers: connections are indexed by actor id.
-        if ld.values.contains_key("config_b64") || ld.values.contains_key("config_json") {
-            let config =
-                ConnectionConfig::new_from(&ld.values, self.fallback_config.cluster_uris.clone())?;
-            let client = self.create_client(config).await?;
-            let mut connections = self.connections.write().await;
-            connections.insert(ld.actor_id.to_string(), client);
-        }
+        // In this multiplexed version of the capability provider, link definitions
+        // contain no data
+
         Ok(true)
     }
 
     /// Handle notification that a link is dropped
-    #[instrument(level = "debug", skip(self))]
+    #[instrument(level = "debug", skip(self), fields(actor_id = ?actor_id))]
     async fn delete_link(&self, actor_id: &str) {
-        let mut connections = self.connections.write().await;
-        let _ = connections.remove(actor_id);
+        // Nothing extra is necessary here since link definitions are not
+        // the unit of correlation to NATS connections
     }
 
     /// Handle shutdown request
+    #[instrument(level = "debug", skip(self))]
     async fn shutdown(&self) -> Result<(), Infallible> {
-        let mut connections = self.connections.write().await;
-        connections.clear();
-
         Ok(())
     }
 }
 
 /// Handle LatticeController methods
 #[async_trait]
 impl LatticeController for LatticeControllerProvider {
-    #[instrument(level = "debug", skip(self, ctx, arg), fields(actor_id = ?ctx.actor))]
+    /// Sets lattice credentials and stores them in the cache to be used to create a
+    /// connection in the next operation that requires one
+    #[instrument(level = "debug", skip(self, ctx, arg), fields(actor_id = ?ctx.actor, lattice_id = ?arg.lattice_id))]
+    async fn set_lattice_credentials(
+        &self,
+        ctx: &Context,
+        arg: &SetLatticeCredentialsRequest,
+    ) -> RpcResult<CtlOperationAck> {
+        self.connections
+            .put_config(
+                &arg.lattice_id,
+                ConnectionConfig {
+                    cluster_uris: vec![arg
+                        .nats_url
+                        .clone()
+                        .unwrap_or_else(|| "0.0.0.0:4222".to_string())],
+                    auth_jwt: arg.user_jwt.clone(),
+                    auth_seed: arg.user_seed.clone(),
+                    lattice_prefix: arg.lattice_id.to_string(),
+                    timeout_ms: 2_000,
+                    auction_timeout_ms: 5_000,
```

Comment:
Should these be configurable?

Reply:
eventually I think so. not sure if it's required for this version since they were hardcoded previously as well.

Reply:
This would be a good "good first issue" to drop in the issue list :)